packages/clerk-js/src/ui/components/SignIn/SignInFactorOneCodeForm.tsx

```diff
@@ -37,23 +37,37 @@ export const SignInFactorOneCodeForm = (props: SignInFactorOneCodeFormProps) =>
   const supportEmail = useSupportEmail();
   const clerk = useClerk();
 
+  const shouldAvoidPrepare = signIn.firstFactorVerification.status === 'verified' && props.factorAlreadyPrepared;
+
   const goBack = () => {
     return navigate('../');
   };
 
-  React.useEffect(() => {
-    if (!props.factorAlreadyPrepared) {
-      prepare();
+  const prepare = () => {
+    if (shouldAvoidPrepare) {
+      return;
     }
-  }, []);
 
-  const prepare = () => {
     void signIn
       .prepareFirstFactor(props.factor)
       .then(() => props.onFactorPrepare())
       .catch(err => handleError(err, [], card.setError));
   };
 
+  useFetch(
+    shouldAvoidPrepare
+      ? undefined
+      : () =>
+          signIn
+            ?.prepareFirstFactor(props.factor)
+            .then(() => props.onFactorPrepare())
+            .catch(err => handleError(err, [], card.setError)),
+    {
+      name: 'prepare',
+      strategy: props.factor.strategy,
+    },
```

Comment:
1. Let's use a more specific name for the cache key
2. If you go back to start and use a different identifier (email) it would not auto-prepare on render.
```suggestion
    {
      name: 'signIn.prepareFirstFactor',
      factor: props.factor,
    },
```

Reply:
Good catch!

Reply:
I've updated it here: https://github.com/clerk/javascript/pull/5095/commits/2c0390594be32c7b143abdc1e8ea7ad983dd378c

The behavior now is the following: It'll cache for a given identifier, so if you go back to `/sign-in` and submit again, it won't send another email cause it's cached - I've added a smaller stale time for that reason, unless you click to "Resend"

When refreshing the page, it'll only fetch once instead of twice (previous bug)